File: server/src/main/java/bit/project/server/entity/Supplier.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import javax.persistence.Id;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Supplier{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private String name;

    private String contact1;

    private String contact2;

    private String fax;

    private String email;

    @Lob
    private String address;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Supplierstatus supplierstatus;

    @ManyToOne
    private Route route;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @JsonIgnore
    @OneToMany(mappedBy = "supplier")
    private List<Collection> supplierCollectionList;


    public Supplier(Integer id){
        this.id = id;
    }

    public Supplier(Integer id, String code, String name){
        this.id = id;
        this.code = code;
        this.name = name;
    }

}

File: client/src/app/entities/supplier.ts
import {User} from './user';
import {Route} from './route';
import {DataPage} from '../shared/data-page';
import {Supplierstatus} from './supplierstatus';

export class Supplier {
  id: number;
  code: string;
  name: string;
  supplierstatus: Supplierstatus;
  route: Route;
  contact1: string;
  contact2: string;
  fax: string;
  email: string;
  address: string;
  description: string;
  creator: User;
  tocreation: string;
}

export class SupplierDataPage extends DataPage{
    content: Supplier[];
}
